images/kafka/kafka-connectors-setup/main.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path"
+
+	jsoniter "github.com/json-iterator/go"
+	archiver "github.com/mholt/archiver/v3"
+	"gopkg.in/alecthomas/kingpin.v2"
+	"gopkg.in/yaml.v2"
+)
+
+// Connector Struct
+type Connector struct {
+	Resources []string    `json:"resources,omitempty" yaml:"resources,omitempty"`
+	Config    interface{} `json:"config,omitempty" yaml:"config,omitempty"`
+}
+
+// ConfigFile Struct
+type ConfigFile struct {
+	Connectors map[string]Connector `json:"connectors,omitempty" yaml:"connectors,omitempty"`
+	Resources  []string             `json:"resources,omitempty" yaml:"resources,omitempty"`
+}
+
+var (
+	app        = kingpin.New("kafka-connectors", "A command-line kafka connectors configuration parser and setup helper.")
+	config     = app.Flag("config", "Path to the configuration file.").Required().File()
+	download   = app.Command("download", "Downloads the resources needed for connectors.")
+	register   = app.Command("register", "Register the connectors in kafka-connect.")
+	endpoint   = register.Arg("endpoint", "Kafka Connect REST endpoint").Required().String()
+	configFile = &ConfigFile{}
+)
+
+func main() {
+	parsed, _ := app.Parse(os.Args[1:])
+	if *config != nil {
+		b, _ := ioutil.ReadAll(*config)
+		log.Printf("Parsing configuration: %s", string(b))
+		if err := yaml.Unmarshal(b, configFile); err != nil {
+			panic(err)
+		}
+		switch parsed {
+		case "register":
+			log.Printf("Registering to endpoint %s", *endpoint)
+			for name, config := range configFile.Connectors {
+				log.Printf("Parsing connector: %s", name)
+				log.Printf("Registering connector configuration: %v\n", *endpoint, config.Config)
+				RegisterConnector(*endpoint, config.Config)
+			}
+
+		case "download":
+			downloadDirectory, _ := os.Getwd()
```

Comment:
can you please extract the logic away from `main`? so we can test this logic easily